fuel-tx/src/transaction/types/input.rs

```diff
@@ -236,13 +236,44 @@ impl AsFieldFmt for PredicateCode {
     derive(fuel_compression::Compress, fuel_compression::Decompress)
 )]
 pub enum Input {
+    // TODO: Consider consolidating all V1 types into single variant.
+    //   This will require us modifying the serialization, but will clean
+    //   up this code significantly.
+    //   https://github.com/FuelLabs/fuel-vm/issues/956
     CoinSigned(CoinSigned),
     CoinPredicate(CoinPredicate),
     Contract(Contract),
     MessageCoinSigned(MessageCoinSigned),
     MessageCoinPredicate(MessageCoinPredicate),
     MessageDataSigned(MessageDataSigned),
     MessageDataPredicate(MessageDataPredicate),
+    InputV2(InputV2),
```

Comment:
We discussed about flattening these to aid with canonical serialization. Is there a reason to have a separate type here? How does it affect serialization size?